grabbit/src/main/groovy/com/twcable/grabbit/spring/batch/repository/GrabbitExecutionContextDao.groovy

```diff
@@ -0,0 +1,24 @@
+package com.twcable.grabbit.spring.batch.repository
+
+import org.springframework.batch.core.repository.dao.ExecutionContextDao
+import org.springframework.batch.item.ExecutionContext;
+
+
+/**
+ * Modified DAO Interface for persisting and retrieving {@link ExecutionContext}
+ * @see ExecutionContextDao for more details
+ */
+interface GrabbitExecutionContextDao extends ExecutionContextDao {
+
+    /**
+     * Returns job execution context paths by comparing "executionId" property on "executionContext/job/<id>" with
+     * "executionId" property on JobExecutions for the @param jobExecutionResourcePaths
+     */
+    public List<String> getJobExecutionContextPaths(List<String> jobExecutionResourcePaths)
```

Comment:
Yeah, I suppose the term "primitives" isn't accurate in this case :-) - rather, "low-level detail objects." It feels sorta off to have such weak coupling points defined on these interfaces. I almost wonder if we can keep some of this exchange in our business objects through type adaptation. It may not make sense in this case -  I'd need to look closer when I'm off my phone, and settled. It may also be asking to dig a deep hole :S hmm, I dunno, just swashbuckling